drivers/eeprom/eeprom_fm24w256.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2021 Caspar Friedrich <c.s.w.friedrich@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <devicetree.h>
+#include <devicetree/gpio.h>
+#include <drivers/eeprom.h>
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+#include <logging/log.h>
+
+#define DT_DRV_COMPAT cypress_fm24w256
+
+#define LEVEL POST_KERNEL
+#define PRIORITY CONFIG_I2C_INIT_PRIORITY
+
+// 256-Kbit ferroelectric random access memory logically organized as 32K × 8 bit
+#define SIZE KB(32)
+
+LOG_MODULE_REGISTER(fm24w256, CONFIG_EEPROM_LOG_LEVEL);
+
+struct config {
+	const struct device *i2c_dev;
+	const uint16_t i2c_addr;
+	const struct gpio_dt_spec wp_spec;
+};
+
+struct data {
+	void *dummy;
+};
+
+static int init(const struct device *dev)
+{
+	int ret;
+
+	const struct config *config = dev->config;
+
+	if (config->wp_spec.port) {
+		ret = gpio_pin_configure_dt(&config->wp_spec, GPIO_OUTPUT_INACTIVE);
+		if (ret) {
```

Comment:
```suggestion
		if (ret < 0) {
```

MISRA-C 2021 rule 14.4 (https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html).